src/index.ts

```diff
@@ -34,14 +33,36 @@ export enum Status {
   NotAcceptable = 406
 }
 
+export interface CookieOptions {
+  expires?: Date
+  domain?: string
+  httpOnly?: boolean
+  maxAge?: number
+  path?: string
+  sameSite?: boolean | 'strict' | 'lax'
+  secure?: boolean
+  signed?: boolean
+}
+
 /**
  * A `Conn`, short for "connection", models the entirety of a connection between the HTTP server and the user agent,
  * both request and response.
  * State changes are tracked by the phanton type `S`
  */
-export class Conn<S> {
+export interface Conn<S> {
   readonly '-S': S
-  constructor(readonly req: express.Request, readonly res: express.Response) {}
+
+  endResponse: () => void
+
+  getBody: () => any
+  getHeader: (name: string) => string | undefined
+  getParams: () => any
+  getQuery: () => any
+
```

Comment:
`io-ts@0.9.5` exports a `mixed` type which is safer than `any`, I'd return `mixed` from all these getters

```ts
getBody: () => t.mixed
getHeader: (name: string) => string | undefined // not sure here
getParams: () => t.mixed
getQuery: () => t.mixed
```

Reply:
koa's `getHeader` equivalent returns `string`, express's is `string | undefined` so I guess that is the best we can do

Reply:
koa's type definition looks wrong, how could a `getHeader` function return `string`? What if the header is not there?

Sorry for the cryptic comment, my "not sure here" was more like a question: "should we use `mixed` here too?" 

I vote for yes because

- the output of `getHeader` must validated anyway (cfr [index.ts/header function](https://github.com/gcanti/hyper-ts/blob/07ffc44a6fae7e1116db3bcdd222bca3cb6aa9bf/src/index.ts#L259))
- is more future proof (let's say another implementation return `string | null`, or a pair name/value `[string, string | undefined]`, etc...)